### ISSUE TITLE
Releases, versioning

### ISSUE THREAD

**Issue by seemk** [open-telemetry/opentelemetry-cpp-contrib#11]
How should we release different contrib projects?

nginx now (#10) builds release binaries for different platforms and I'd like expose these artifacts via Github releases.
Should each project have it's own version tag? E.g. `nginx-0.1.0` and `httpd-0.1.0` so when the tag is created a GH release will be available.

**Comment by lalitb** [comment]
> Should each project have it's own version tag? E.g. nginx-0.1.0 and httpd-0.1.0 so when the tag is created a GH release will be available.

I will vote for this. Each project within repo should have separate version tags and releases.

**Comment by seemk** [comment]
If we are ok with this, I'll do a release for `nginx-0.1.0`, it will provide the nginx module binaries as part of the release.

**Comment by lalitb** [comment]
Looks good to me.

**Comment by seemk** [comment]
Looks like I can't do it in this repo - no `Create a new release` button available :slightly_frowning_face: Access rights I presume

**Comment by lalitb** [comment]
Ok, I think @maxgolov is the admin of this repo and can give you write access to create a release. And probably I am getting write access through the `cpp-contrib-approvers` group. I did try adding you to the group earlier but seems need to be a member of opentelemetry org for that.
You can raise a ticket to be a member (example) : https://github.com/open-telemetry/community/issues/647 . I can be one of the sponsors ( we need two ). Once this is done, I can add you to the approver's group. Hopefully, that should give you permission to create a release.

**Comment by seemk** [comment]
Thanks! Created a request: https://github.com/open-telemetry/community/issues/699

**Comment by lalitb** [comment]
Have added you to `cpp-contrib-approvers`. Don't think if that will help, need to wait for @maxgolov next week for right permissions.

**Comment by maxgolov** [comment]
@seemk @lalitb

I vaguely remember that when we initially got it set up, we had to ask for the open-telemetry org admins to set permissions on it.. Can I help creating a tag? Would you be able to provide me with release notes, and how do you want to label it - `v0.1.0` , or do we align it with the mainline tag?

Let's discuss it on slack.

**Comment by maxgolov** [comment]
I created the tag `v0.1.0`. Hope that it addresses your immediate needs.

**Comment by TomRoSystems** [comment]
Yes, thanks. We would like to include also some artefacts (assets) as well during this process. I've contacted @SergeyKanzhelev and he suggested for me to file an issue which I did - https://github.com/open-telemetry/community/issues/718

**Comment by maxgolov** [comment]
@TomRoSystems - once you get permissions, you may be able to attach additional artifacts. It would be great though if we somehow script the creation of binaries. That is to avoid attaching prebuilt binaries that have not been created by the GitHub Actions, or some other `secure` process. Attaching source or documentation in any form should be fine.

**Comment by SergeyKanzhelev** [comment]
Knowing GH permissions nonsense, if somebody will need to get Admin on repo to set everything up - let me know (ideally by adding details in open-telemetry/community#718).

**Comment by seemk** [comment]
@maxgolov Thanks for this, and my bad for not noticing this discussion here earlier :slightly_smiling_face: 

We decided to tag different instrumentation releases with each their own tag, e.g. `nginx-0.1.0`, but I can edit these now.

I was hoping the artifacts would show up under the releases, but CI will need to package and publish them separately (via https://github.com/softprops/action-gh-release), going to add it soon.
